client/src/main/java/org/apache/celeborn/client/read/LocalPartitionReader.java

```diff
@@ -51,7 +51,7 @@
 public class LocalPartitionReader implements PartitionReader {
 
   private static final Logger logger = LoggerFactory.getLogger(LocalPartitionReader.class);
-  private static volatile ThreadPoolExecutor readLocalShufflePool;
+  private volatile ThreadPoolExecutor readLocalShufflePool;
```

Comment:
This is expected to be a static variable, not instance.

Reply:
@mridulm, this static variable does not satisfy [StaticAssignmentInConstructor](https://errorprone.info/bugpattern/StaticAssignmentInConstructor).

Reply:
That is a reasonable recommandation when sufficient care is not taken to address the issues.
Here, we are doing the right thing by making sure it is initialized in a safe manner the first time the intiial instance of the class is created

Also note that the current change is a regression - for each instance of `LocalPartitionReader` we will create a new thread pool.